web/modules/custom/study_blocks/src/Plugin/Block/DefaultBlock.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\study_blocks\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'DefaultBlock' block.
+ *
+ * @Block(
+ *  id = "default_block",
+ *  admin_label = @Translation("Default block"),
+ * )
+ */
+class DefaultBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+
+  protected $MyEntityStorage;
+  protected $NodeStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityStorageInterface $entity_storage, EntityStorageInterface $node_storage)
+  {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->MyEntityStorage = $entity_storage;
+    $this->NodeStorage = $node_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)
+  {
+    $storage = $container->get('entity.manager');
+    return new static(
+      $configuration, $plugin_id, $plugin_definition,
+      $storage->getStorage('default_entity'),
```

Comment:
inject services instead of storages


```
protected $entity_type_manager;

public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
    parent::__construct($configuration, $plugin_id, $plugin_definition);
    $this->entity_type_manager = $entity_type_manager;
}

public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
  return new static(
    $configuration, 
    $plugin_id, 
    $plugin_definition,
    $container->get('entity_type.manager')
  );
}

...

public function build() {
  ...
  $node_storage = $this->entity_type_manager->getStorage('node');
  $default_entity_storage = $this->entity_type_manager->getStorage('default_entity');
  ...
}
```